simulator/src/main/java/com/hedera/block/simulator/BlockStreamSimulatorApp.java

```diff
@@ -53,15 +58,73 @@ public BlockStreamSimulatorApp(
             @NonNull Configuration configuration,
             @NonNull BlockStreamManager blockStreamManager,
             @NonNull PublishStreamGrpcClient publishStreamGrpcClient) {
-        this.configuration = configuration;
         this.blockStreamManager = blockStreamManager;
         this.publishStreamGrpcClient = publishStreamGrpcClient;
 
         blockStreamConfig = configuration.getConfigData(BlockStreamConfig.class);
-
         delayBetweenBlockItems = blockStreamConfig.delayBetweenBlockItems();
     }
 
+    /**
+     * Constructor that accepts a custom configuration and initializes dependencies internally.
+     *
+     * @param configuration the custom configuration to use
+     */
+    public BlockStreamSimulatorApp(Configuration configuration) {
+        this(configuration, createComponent(configuration));
+    }
+
+    /**
+     * Private constructor that initializes the class using a configuration and a Dagger component.
+     *
+     * @param configuration the configuration to use
+     * @param component the Dagger component for dependency injection
+     */
+    private BlockStreamSimulatorApp(
+            Configuration configuration, BlockStreamSimulatorInjectionComponent component) {
+
+        this(
+                configuration,
+                component.getBlockStreamManager(),
+                component.getPublishStreamGrpcClient());
+    }
+
+    /**
+     * Default constructor that loads the default configuration and initializes dependencies.
+     *
+     * @throws IOException if an I/O error occurs while loading the default configuration
+     */
+    public BlockStreamSimulatorApp() throws IOException {
+        this(loadDefaultConfiguration());
+    }
+
+    /**
+     * Helper method to create the Dagger component with the given configuration.
+     *
+     * @param configuration the configuration to use
+     * @return the Dagger component
+     */
+    private static BlockStreamSimulatorInjectionComponent createComponent(
+            Configuration configuration) {
+        return DaggerBlockStreamSimulatorInjectionComponent.factory().create(configuration);
+    }
+
+    /**
+     * Loads the default configuration for the simulator app.
+     *
+     * @return the default configuration
+     * @throws IOException if an I/O error occurs while loading the configuration
+     */
+    private static Configuration loadDefaultConfiguration() throws IOException {
+        ConfigurationBuilder configurationBuilder =
+                ConfigurationBuilder.create()
+                        .withSource(SystemEnvironmentConfigSource.getInstance())
```

Comment:
This code looks like it's from `Server.java`. We're going to change how we bring in environment variables. Perhaps coordinate with Atanas to make the same changes here